glean-core/src/debug.rs

```diff
@@ -33,63 +39,77 @@ pub struct DebugOptions {
     pub log_pings: DebugOption<bool>,
     /// Option to add the X-Debug-ID header to every ping request.
     pub debug_view_tag: DebugOption<String>,
+    /// Option to add the X-Source-Tags header to ping requests. This will allow the data
+    /// consumers to classify data depending on the applied tags.
+    pub sourge_tags: DebugOption<Vec<String>>,
 }
 
 impl DebugOptions {
     pub fn new() -> Self {
         Self {
-            log_pings: DebugOption::new(GLEAN_LOG_PINGS, None),
-            debug_view_tag: DebugOption::new(GLEAN_DEBUG_VIEW_TAG, Some(validate_debug_view_tag)),
+            log_pings: DebugOption::new(GLEAN_LOG_PINGS, get_bool_from_str, None),
+            debug_view_tag: DebugOption::new(GLEAN_DEBUG_VIEW_TAG, identity, Some(validate_tag)),
+            sourge_tags: DebugOption::new(
+                GLEAN_SOURCE_TAGS,
+                tokenize_string,
+                Some(validate_source_tags),
+            ),
         }
     }
 }
 
 /// A representation of a debug option,
 /// where the value can be set programmatically or come from an environment variable.
 #[derive(Debug)]
-pub struct DebugOption<T, F = fn(T) -> Option<T>> {
+pub struct DebugOption<T, E = fn(String) -> Option<T>, F = fn(T) -> bool> {
```

Comment:
Generics don't _have_ to eb single letter. At some point we could consider making them more expressive.
`F` here definitely doesn't capture anything useful about this.

Reply:
@badboy any suggestion or something I should tackle here? Do you want me to file a bug?

Reply:
What about using `V` instead? Or `VALDN` / `Valdn` or `Validation`?